SMLHelper/Options/ModColorOption.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using UnityEngine;
+using UnityEngine.Events;
+
+namespace SMLHelper.Options
+{
+    /// <summary>
+    /// Contains all the information about a color changed event.
+    /// </summary>
+    public class ColorChangedEventArgs : ConfigOptionEventArgs<Color>
+    {
+        /// <summary>
+        /// Constructs a new <see cref="ToggleChangedEventArgs"/>.
+        /// </summary>
+        /// <param name="id">The ID of the <see cref="ModColorOption"/> that was changed.</param>
+        /// <param name="value">The new value for the <see cref="ModColorOption"/>.</param>
+        public ColorChangedEventArgs(string id, Color value) : base(id, value) { }
+    }
+
+    /// <summary>
+    /// A basic mod option class for handling an option that can be any <seealso cref="Color"/>.
+    /// </summary>
+    public class ModColorOption : ModOption<Color, ColorChangedEventArgs>
```

Comment:
Can we get a Bepinex Config Entry conversion extension added.
If you want let me know and I will do it.